### ISSUE TITLE
[BUG]: finalize_model() does not train the model on the entire dataset if test_data is used in the setup

### ISSUE THREAD

**Issue by cespos** [pycaret/pycaret#2717]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [ ] I have confirmed this bug exists on the develop branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@develop).

### Issue Description

Hi,

While doing some tests, I have noticed that the finalize_model() function of pycaret.classification behaves correctly if train_size is used in the setup. However, when I used test_data in the setup, the finalized model was not trained on the entire dataset (train+test) but only on the training set. I checked this by looking at the size of the training set after tune_model() and after finalize_model(). And I confirmed this also by looking at the predictions returned by the tuned and final models. With train_size, the predictions of tuned and final model were different, whereas with test_data, they were the same.

I am attaching here a figure to show this.
<img width="687" alt="pycaret_bug_finalize_model" src="https://user-images.githubusercontent.com/31316943/177497981-8223414a-79f8-4e49-a2f9-f26a9bd464bd.PNG">

Thanks,
Carmen

### Reproducible Example

```python
from pycaret.datasets import get_data
from pycaret.classification import *

dataset = get_data('credit').iloc[0:1000]

data = dataset.sample(frac=0.95, random_state=786)
data_unseen = dataset.drop(data.index)
data.reset_index(inplace=True, drop=True)
data_unseen.reset_index(inplace=True, drop=True)


# with default train_size=0.7
exp_clf101 = setup(data = data, target = 'default', session_id=123) 

best_model = compare_models(['rf', 'xgboost'])
best_model.oob_score=True
best_model1 = tune_model(best_model, n_iter=50, choose_better=True, optimize='MCC')
best_model1_final = finalize_model(best_model1)

print(data.shape)
print(best_model1.oob_decision_function_.shape)
print(best_model1_final.oob_decision_function_.shape)

preds_tune = predict_model(
                best_model1,
                data=data_unseen,
                raw_score=True,
            )

preds_finale = predict_model(
                best_model1_final,
                data=data_unseen,
                raw_score=True,
            )

print(preds_tune[['Label'] + [s for s in list(preds_tune) if 'Score' in s]].head())
print(preds_finale[['Label'] + [s for s in list(preds_finale) if 'Score' in s]].head())

# using test_data in the setup
exp_clf101 = setup(data = data, target = 'default', session_id=123, test_data = data_unseen)
best_model_test_data = create_model('rf')
best_model_test_data.oob_score = True
best_model_test_data1 = tune_model(best_model_test_data, n_iter=3, choose_better=True, optimize='MCC')
best_model_test_data1_final = finalize_model(best_model_test_data1)

print(data.shape)
print(best_model_test_data1_final.oob_decision_function_.shape)
print(best_model_test_data1.oob_decision_function_.shape)

preds_tune2 = predict_model(
                best_model_test_data1,
                data=data_unseen,
                raw_score=True,
            )

preds_finale2 = predict_model(
                best_model_test_data1_final,
                data=data_unseen,
                raw_score=True,
            )

print(preds_tune2[['Label'] + [s for s in list(preds_tune2) if 'Score' in s]].head())
print(preds_finale2[['Label'] + [s for s in list(preds_finale) if 'Score' in s]].head())
```

### Expected Behavior

finalize_model() should train the model on the entire train+test set even when test_data is used in the setup.

### Actual Results

```python-traceback
I have attached a figure in the Issue Description
```

### Installed Versions

<details>
2.3.0
</details>

**Comment by cespos** [comment]
It seems like it is working in the development branch. 
Do you know already when the new version will be released?
![image](https://user-images.githubusercontent.com/31316943/178038319-ebc1a35e-42a0-4843-9a1e-5f76239c922b.png)

**Comment by tvdboom** [comment]
We don't have an exact date yet. Updates are a bit slow now because of holidays, but we are aiming to make a release before the end of the summer.
